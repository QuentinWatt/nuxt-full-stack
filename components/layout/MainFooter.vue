<script setup lang="ts">
  const date = new Date();
</script>

<template>
  <footer class="bg-gray-800 py-6">
    <div class="container mx-auto px-3">
      <div class="flex justify-between items-center">
        <span class="text-2xl font-bold text-white h-12">MyLogo</span>
        <span class="text-sm text-gray-300">&copy; {{ date.getFullYear() }}</span>
      </div>
    </div>
  </footer>
</template>