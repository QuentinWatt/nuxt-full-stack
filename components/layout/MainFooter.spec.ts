import { expect, test } from "vitest";
import MainFooter from "./MainFooter.vue";
import { mountSuspended } from "@nuxt/test-utils/runtime";

test("It has a copyright with the current year", async () => {
  const component = await mountSuspended(MainFooter);
  expect(component.html()).toContain("Â© 2024");
});

test("It has a logo", async () => {
  const component = await mountSuspended(MainFooter);
  expect(component.html()).toContain("MyLogo");
});

test("It matches the snapshot", async () => {
  const component = await mountSuspended(MainFooter);
  expect(component.html()).matchSnapshot();
});
