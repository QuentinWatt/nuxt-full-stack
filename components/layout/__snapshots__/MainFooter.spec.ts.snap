// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`It matches the snapshot 1`] = `
"<footer class="bg-gray-800 py-6">
  <div class="container mx-auto px-3">
    <div class="flex justify-between items-center"><span class="text-2xl font-bold text-white h-12">MyLogo</span><span class="text-sm text-gray-300">Â© 2024</span></div>
  </div>
</footer>"
`;
